tree-sitter-js/README

```diff
@@ -0,0 +1,13 @@
+# tree-sitter 0.22.6
+
+tree-sitter.js and tree-sitter.wasm are imported from
+https://github.com/tree-sitter/tree-sitter/releases/tag/v0.22.6
+
+# tree-sitter-css 0.21.0
+
+tree-sitter-css.wasm is generated by the following commands in empty directory:
+```
+npm install --save-dev tree-sitter-cli tree-sitter-css
+npx tree-sitter build --wasm node_modules/tree-sitter-css
+```
```

Comment:
Given that we have a provisioning step ([where we already install node](https://github.com/mozsearch/mozsearch/blob/0f89a38f08c8cdc4375291feaf6ee11c5736961a/infrastructure/indexer-provision.sh#L23-L46)) and we have a build step, I don't understand why we're introducing manual steps here that result in us having to check in non-trivial binary blobs that would potentially beget additional manual work in the future to update.

This would potentially go with using cssparser as well.

If this is motivated by how long it can take to provision or the build times of the `make build-test-repo`, I think it probably makes more sense to try and improve those to do less re-work or otherwise be more ergonomic[1].  Right now I think the current idiom for experimenting with adding things to the provisioning process and being able to do try runs is to update https://github.com/mozsearch/mozsearch/blob/master/infrastructure/indexer-update.sh so that the indexer will run the steps at its startup.  Also one can obviously run the script from within the provisioned docker image.  (But if you reprovision, the script will even get run twice, so it will have effects and we will verify that it won't break when an indexer AMI updates.)

Aside: I do understand it's not unusual for JS projects to potentially vendor WASM, but I think that tends to be justified by environments where there is already aggressive vendoring and/or the inability/policy to not run additional build steps beyond webpack/esbuild or whatever.  (I know we have our weird/ugly sax.js thing, but that's definitely not best practice for anything! ;)

1: Like Nicolas had added a [nix flake](https://github.com/mozsearch/mozsearch/blob/master/flake.nix) that I don't know if that would be a better workflow for us to use in some cases?  It seems like the flake may only be for outside of docker, though, which makes sense since we don't install nix into the docker image.

Reply:
Relatedly, I think it would be nice to avoid adding new top-level directories like this where it seems like without some kind of hierarchy.  That is, if you ended up adding another WASM dep for js-analyze, it seems like it would also end up as another top-level directory too.  Maybe we could add a subdir like `scripts/js-analyze-deps` or something to be the parent of js-analyze specific things which would also help make the relationship more clear?

Alternately, maybe given the significance and notability of js-analyze, maybe we should create a new root dir like `web-analyze` and `js-analyze.js` the script could move in there and maybe be renamed to `web-analyze.js` (maybe under a `src/` subdir?) since it covers all of js/css/html.  And then the dep subdirs could live there too.